core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorMetadata.java

```diff
@@ -432,6 +432,17 @@ default void addColumn(ConnectorSession session, ConnectorTableHandle tableHandl
         throw new TrinoException(NOT_SUPPORTED, "This connector does not support adding columns");
     }
 
+    /**
+     * Add the specified field, potentially nested, to a row.
+     *
+     * @param parentPath path to a field within the column, without leaf field name.
+     */
+    @Experimental(eta = "2023-06-01") // TODO add support for rows inside arrays and maps and for anonymous row fields
+    default void addField(ConnectorSession session, ConnectorTableHandle tableHandle, List<String> parentPath, String fieldName, Type type)
```

Comment:
do we want to pass `ignoreExisting` into the connector?

(we don't do that for addColumn, but i think we should, since only the connector can do things atomically, engine cannot)